### PR TITLE
feat: Request claims required Properties and approved attesters

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes https://github.com/KILTprotocol/ticket/issues/1199

The request-claims-for-ctype message currently only allows specifying acceptable ctypes.

To allow more use-cases, expand on it to be able to specify per ctype:

- [x] required attributes
- [x] accepted attesters

## How to test:

Added methods to verify the required properties. 

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/messaging/src/Message.utils.ts [hunk 1]
```diff
@@ -15,12 +15,41 @@ import {
...
+ * @throws [[ERROR_CTYPE_HASH_NOT_PROVIDED]] when the properties do not match the provide [[CType]].
+ *
+ * @returns Returns the properties back.
+ */
+
+export function verifyRequiredCTypeProperties(
+  requiredProperties: string[],
+  cType: ICType
+): boolean {
+  if (!cType.hash) throw SDKErrors.ERROR_CTYPE_HASH_NOT_PROVIDED()
```
Comment (LeonFLK) [inline packages/messaging/src/Message.utils.ts:39]: Shouldn't we use our isICType typeguard instead of just checking the hash?

**Review by LeonFLK — COMMENTED** [review]
Small nitpick, use isICType type guard instead of just checking existence of hash.
Or is this not feasible because the CType is partial, and thus not passing the type guard?

File: packages/messaging/src/Message.utils.ts (hunk 1 above)
Comment (Dudleyneedham) [reply packages/messaging/src/Message.utils.ts]: I agree, I have updated it.

**Review by LeonFLK — APPROVED** [review]
LGTM

**Comment by LeonFLK** [comment]
Integration test are expected to fail, as they are broken momentarily.

**Pull request merged**